flash/graph/GraphClassification/data.py

```diff
@@ -0,0 +1,380 @@
+                transform=valid_transform,
+            ) if valid_filepaths is not None else None
+        )
+
+        test_ds = (
+            FilepathDataset(
+                filepaths=test_filepaths,
+                labels=test_labels,
+                loader=loader,
+                transform=valid_transform,
+            ) if test_filepaths is not None else None
+        )
+
+        return cls(
+            train_ds=train_ds,
+            valid_ds=valid_ds,
+            test_ds=test_ds,
+            batch_size=batch_size,
+            num_workers=num_workers,
+        )
+
+    @classmethod
+    def from_folders(
+        cls,
+        train_folder: Optional[Union[str, pathlib.Path]],
+        train_transform: Optional[Callable] = None,
+        valid_folder: Optional[Union[str, pathlib.Path]] = None,
+        valid_transform: Optional[Callable] = None,
+        test_folder: Optional[Union[str, pathlib.Path]] = None,
+        loader: Callable = torch.load,
+        batch_size: int = 4,
+        num_workers: Optional[int] = None,
+        **kwargs
+    ):
+        """
+        Creates a GraphClassificationData object from folders of Graphs arranged in this way: ::
+
+            train/dog/xxx.png
+            train/dog/xxy.png
+            train/dog/xxz.png
+            train/cat/123.png
+            train/cat/nsdf3.png
+            train/cat/asd932.png
+
+        Args:
+            train_folder: Path to training folder.
+            train_transform: Graph transform to use for training set.
+            valid_folder: Path to validation folder.
+            valid_transform: Graph transform to use for validation and test set.
+            test_folder: Path to test folder.
+            loader: A function to load an Graph given its path.
+            batch_size: Batch size for data loading.
+            num_workers: The number of workers to use for parallelized loading.
+                Defaults to None which equals the number of available CPU threads.
+
+        Returns:
+            GraphClassificationData: the constructed data module
+
+        Examples:
+            >>> img_data = GraphClassificationData.from_folders("train/") # doctest: +SKIP
+
+        """
+        train_ds = FlashDatasetFolder(train_folder, transform=train_transform, loader=loader)
+        valid_ds = (
+            FlashDatasetFolder(valid_folder, transform=valid_transform, loader=loader)
+            if valid_folder is not None else None
+        )
+
+        test_ds = (
+            FlashDatasetFolder(test_folder, transform=valid_transform, loader=loader)
+            if test_folder is not None else None
+        )
+
+        datamodule = cls(
+            train_ds=train_ds,
+            valid_ds=valid_ds,
+            test_ds=test_ds,
+            batch_size=batch_size,
+            num_workers=num_workers,
+        )
+
+        datamodule.num_classes = len(train_ds.classes)
+        datamodule.data_pipeline = GraphClassificationDataPipeline(
```

Comment:
Where is it defined ?

Reply:
I have included it, although will most likely need to be changed since it is mostly the ImageClassificationPipeline